llvm/include/llvm/Support/DXILABI.h

```diff
@@ -90,4 +90,9 @@ enum class ElementType : uint32_t {
 } // namespace dxil
 } // namespace llvm
 
+// Generate a unique value for given Major, Minor pair of Shader Model
+// version. Allows for 100 minor versions for a given major version number.
+// To be used uniformly by DXILEmitter backend as well as DXIL Lowering pass.
+#define COMPUTE_SM_VERSION_VALUE(MAJ, MIN) ((MAJ * 100) + MIN)
```

Comment:
Is there a reason we're munging these into one value?  It seems that if we can keep them separate without having to encode them in some way things might be simpler.

Reply:
It seems like we should use `llvm::VersionTuple` for this instead of rolling our own way to mash version numbers together and compare them.